challenge-20/challenge-20.js

```diff
@@ -1,107 +1,128 @@
-/*
-1. Envolva todo o conteúdo desse desafio em uma IIFE.
-2. Adicione a diretiva 'use strict';
-3. Passe por parâmetro para a IIFE os objetos window e document.
-4. Dessa vez não é necessário criar um HTML. Ele já existe, e vamos utilizar
-a marcação criada nele para fazer nosso desafio ;)
+(function (win, doc) {
+  'use strict';
 
-O HTML NÃO PODE ser alterado!
-*/
 
-/*
-Ao carregar a página, pergunte ao usuário "Qual o seu nome?". Atribua o
-resultado à uma variável chamada `username`. Se o usuário não digitar um
-nome, `username` deve receber "Desconhecido".
-Com a resposta, mostre um alert com a mensagem "Bem vindo [USERNAME]!"
-*/
-// ?
+  /*
+  1. Envolva todo o conteúdo desse desafio em uma IIFE.
+  2. Adicione a diretiva 'use strict';
+  3. Passe por parâmetro para a IIFE os objetos window e document.
+  4. Dessa vez não é necessário criar um HTML. Ele já existe, e vamos utilizar
+  a marcação criada nele para fazer nosso desafio ;)
 
-/*
-Agora, pergunte ao usuário "Qual o seu e-mail?", atribuindo o resultado à
-uma variável chamada `email`.
-*/
-// ?
+  O HTML NÃO PODE ser alterado!
+  */
 
-/*
-- Selecione o input de "Nome", atribuindo-o à uma variável chamada
-`$inputUsername`.
-*/
-// ?
+  /*
+  Ao carregar a página, pergunte ao usuário "Qual o seu nome?". Atribua o
+  resultado à uma variável chamada `username`. Se o usuário não digitar um
+  nome, `username` deve receber "Desconhecido".
+  Com a resposta, mostre um alert com a mensagem "Bem vindo [USERNAME]!"
+  */
+  var username = prompt('Qual o seu nome?');
+  (username) ? alert('Bem vindo ' + username ) : username = 'Desconhecido';
```

Comment:
-1
Ternário você só deve usar quando for atribuir o resultado dele a uma variável ou quando for fazer o retorno de uma função. Em outros casos, use o `if`.

Nessa validação, ele nunca mostra a mensagem `Bem vindo Desconhecido`. Deveria mostrar :)